packages/x-charts-pro/src/internals/material/components/BasePopper.tsx

```diff
@@ -0,0 +1,140 @@
+import Paper from '@mui/material/Paper';
+import { ChartBasePopperProps } from '../../slots/chartBaseSlotProps';
+
+function clickAwayWrapper(props: ChartBasePopperProps, content: any) {
+  if (props.onClickAway === undefined) {
+    return content;
+  }
+  return (
+    <ClickAwayListener
+      onClickAway={props.onClickAway as any}
+      touchEvent={props.clickAwayTouchEvent}
+      mouseEvent={props.clickAwayMouseEvent}
+    >
+      {content}
+    </ClickAwayListener>
+  );
+}
+
+function focusTrapWrapper(props: ChartBasePopperProps, content: any) {
+  if (props.focusTrap === undefined) {
+    return content;
+  }
+  return (
+    <MUIFocusTrap open disableEnforceFocus disableAutoFocus>
+      <div tabIndex={-1}>{content}</div>
+    </MUIFocusTrap>
+  );
+}
+
+function wrappers(props: ChartBasePopperProps, content: any) {
+  return focusTrapWrapper(props, clickAwayWrapper(props, content));
+}
+
+const transformOrigin = {
+  'bottom-start': 'top left',
+  'bottom-end': 'top right',
+};
+
+export function BasePopper(props: ChartBasePopperProps) {
+  const {
+    ref,
+    open,
+    children,
+    className,
+    clickAwayTouchEvent,
+    clickAwayMouseEvent,
+    flip,
+    focusTrap,
+    onExited,
+    onClickAway,
+    onDidShow,
+    onDidHide,
+    id,
+    target,
+    transition,
+    placement,
+    ...rest
+  } = props;
+
+  const modifiers = React.useMemo(() => {
+    const result: PopperProps['modifiers'] = [
+      {
+        name: 'preventOverflow',
+        options: {
+          padding: 8,
+        },
+      },
+    ];
+    if (flip) {
+      result.push({
+        name: 'flip',
+        enabled: true,
+        options: {
+          rootBoundary: 'document',
+        },
+      });
+    }
+    if (onDidShow || onDidHide) {
+      result.push({
+        name: 'isPlaced',
+        enabled: true,
+        phase: 'main' as const,
+        fn: () => {
+          onDidShow?.();
+        },
+        effect: () => () => {
+          onDidHide?.();
+        },
+      });
+    }
+    return result;
+  }, [flip, onDidShow, onDidHide]);
+
+  let content: any;
+  if (!transition) {
+    content = wrappers(props, children);
+  } else {
+    const handleExited = (popperOnExited: (() => void) | undefined) => (node: HTMLElement) => {
+      if (popperOnExited) {
+        popperOnExited();
+      }
+
+      if (onExited) {
+        onExited(node);
+      }
+    };
+
+    content = (p: any) =>
+      wrappers(
+        props,
+        <Grow
+          {...p.TransitionProps}
+          style={{ transformOrigin: transformOrigin[p.placement as keyof typeof transformOrigin] }}
+          onExited={handleExited(p.TransitionProps?.onExited)}
+        >
+          <Paper>{children}</Paper>
+        </Grow>,
+      );
+  }
+
+  return (
+    <Popper
+      id={id}
+      className={className}
+      open={open}
+      anchorEl={target as any}
+      transition={transition}
+      placement={placement}
+      modifiers={modifiers}
+      {...rest}
```

Comment:
TODO: We're missing the `material` prop that [the Data Grid has](https://github.com/mui/mui-x/blob/7966bf543079e9b62d8d95046a005481feb7b1f3/packages/x-data-grid/src/material/augmentation.ts#L23)

Reply:
Planning on tackling this as a follow-up. 

We also need to implement base slot props augmentation similar to the data grid.

Reply:
Sure, I'll check it 👍

Reply:
My understanding is that the `material` prop is used to specify props that are specific to Material UI. The props from the "base" components are supposed to be our prop and if the user updates Material UI, we need to ensure compatibility (within the Material UI versions we support). With the `material` prop, the props are directly passed to Material UI, so if a user updates Material UI, they need to handle the breaking changes. 

[Source](https://mui-org.slack.com/archives/C05SBSU8NPR/p1749564375195939)

So we probably want to implement it so users have access to the whole material API.